DigitalLearningSolutions.Web.Tests/ViewModels/MyAccount/MyAccountEditDetailsFormDataTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace DigitalLearningSolutions.Web.Tests.ViewModels.MyAccount
+{
+    using System.Collections.Generic;
+    using System.ComponentModel.DataAnnotations;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Tests.TestHelpers;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.ViewModels.MyAccount;
+    using FluentAssertions;
+    using NUnit.Framework;
+
+    public class MyAccountEditDetailsFormDataTests
+    {
+        [Test]
+        public void
+            MyAccountEditDetailsFormData_Validate_returns_validation_results_for_AllCentreSpecificEmailsDictionary()
+        {
+            // Given
+            var userAccount = UserTestHelper.GetDefaultUserAccount();
+            var allCentreSpecificEmailsDictionary = new Dictionary<string, string?>
+            {
+                { "1", "email@centre1.com" },
+                { "2", "email @centre2.com" },
+                { "3", "email_centre3" },
+                { "4", $"email@centre4.com{new string('m', 300)}" },

```

Comment:
Maybe it's worth making this email 256 characters to test it applies the error on the intended character length cut-off?

Reply:
True, fine as is then 👍